src/Language/Fortran/Parser.hs

```diff
@@ -253,6 +254,18 @@ f77lIncludes incs mods fn bs = do
                                  pf''
         return pf'''
 
+-- | Entry point for include files
+-- 
+-- We can't perform full analysis (though it might be possible to do in future)
+-- but the AST is enough for certain types of analysis/refactoring
+f77lIncIncludes
+  :: [FilePath] -> String -> B.ByteString -> IO [Block A0]
+f77lIncIncludes incs fn bs =
+  case makeParserFixed F77.includesParser Fortran77Legacy fn bs of
+    Left e -> liftIO $ throwIO e
+    Right bls ->
```

Comment:
`Parser`s spit out a `ProgramFile`, so this is misleading. The function can return a `[Block A0]` at the end because uniplate applies its biplate magic.

Reply:
I don't think that's true? `makeParseFixed` returns whatever it's first argument returns, which in the case of `includesParser` is `[Block A0]`.

Reply:
At the very least I've tried running this entry point on include files and it works, which it wouldn't if it were expecting to parse `ProgramUnit`s